src/components/modules/shared/ModuleWrapper.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+export function withParsedProps<ModuleProps>(
+    Module: React.FC<ModuleProps>,
+    validate: (props: unknown) => props is ModuleProps,
+): React.FC<unknown> {
+    const WrappedModule = (props: unknown) => {
+        if (validate(props)) {
+            return <Module {...props} />;
+        }
+        return null;
```

Comment:
I wonder if we want sentry logging here? Or perhaps we just need to catch this with tests (before or after merge)